ngsild/ngsild_connector.py

```diff
@@ -39,11 +19,29 @@ def get_url(self):
         return url
 
     def send_data_array(self, json_object):
+        return_info = []
         d = json.loads(json_object)
         d = d if type(d) is list else [d]
 
         for elem in d:
-            rc, r = c.send_data(json.dumps(elem))
+            try:
+                rc, r = self.send_data(json.dumps(elem))
+                return_info.append({"id": elem['id'],
+                                   "status_code": rc,
+                                   "reason": r})
+            except TypeError as e:
```

Comment:
I suggest raising the exception due to in the server:parser we are controlling the different exceptions that we have in the server (lines 158-180).

I suggest creating a new 400 - 499 status code. I suggest using 422 Unprocessable Entity (status.HTTP_422_UNPROCESSABLE_ENTITY).

Reply:
LGTM